zulipterminal/model.py

```diff
@@ -854,6 +854,23 @@ def fetch_message_history(
         display_error_if_present(response, self.controller)
         return list()
 
+    def fetch_message_read_receipt_user_ids(
+        self, message_id: int
+    ) -> Union[None, str, List[int]]:
+        """
+        Fetches user IDs of users who've read a message using its ID.
+        If fails to fetch due to being unable to retrieve ZFL, returns None
+        """
+        if self.server_feature_level is not None and self.server_feature_level >= 137:
+            response = self.client.call_endpoint(
+                f"/messages/{message_id}/read_receipts", method="GET"
+            )
+            if response["result"] == "success":
+                return response["user_ids"]
+        else:
+            raise RuntimeError("Unsupported server feature level.")
+        return None
```

Comment:
It is a little confusing to place this line here, since one has to trace back to identify which `if` statement might fail to get here, as the previous block is an `if`/`else` pair. For multiple failing situations, this would be more reasonable, but it's better to pair it with the matching `if` if there is only one.

An alternative is to handle the quicker/simpler error case(s) first; this also means the main flow of code is less indented. Since you have test coverage, exploring this should be safe :)

Reply:
Just to confirm, you're talking about the `return None` line, right? How bout we just remove it? I think the runtime error takes care of every invalid case. Plus we aren't handling the case when (if) it returns `None`

Reply:
Yes, that was what I was referring to (unfortunately GitHub doesn't mark the line I click when starting to review inline :/)

Reply:
What you have here is:
```
if continue_after_quick_check:
  do_network_call()
  if network_call_success:
    return success_value
else: # quick check failed
  raise quick_check_error()
return None # network call error
```
I first suggested the `None` might be clearer paired with the `if network_call_success` in a matching `else`, ie. moving the last line in this simplified example just after the fourth.

The alternative of checking for quick errors first is more like:
```
if not continue_after_quick_check:
  raise quick_check_error() # quick check failed

do_network_call()
if network_call_success:
  return success_value
return None # network call error (could be in an 'else')
```